string map_name
---
bool success
---
